### PR TITLE
Dockerizing the app.

### PR DESCRIPTION
**Pull request by j4rs**
Hey @prajaktar, here's a small contribution. Regarding the code I will send my comments that we can discuss  over this PR.

### CODE REVIEW

**Review by prajaktar — APPROVED** [review]

**Comment by prajaktar** [comment]
The enhancements look good to me, I am reading more about docker to understand it. I have changed the base branch from master to develop.

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,6 @@
 
 /config/secrets.yml
 /config/database.yml
```
Comment (j4rs) [inline .gitignore:20]: @prajaktar Why database and secrets are ignored and you are using a different version of them?

Comment (j4rs) [reply .gitignore:20]: Actually I just found it by now looking at the PR and surprised cause they were not committed.

Comment (prajaktar) [reply .gitignore:20]: I had pushed a sample database and secrets file named as database-dist.yml and secrets-dist.yml and ignored secrets.yml and database.yml.

For example, in a team every developer may have different local database_name, username and password. So need to commit this file every time. To start your application, developer can create a copy of database-dist.yml as database.yml and edit the required information as per local credentials.

With docker I suppose this case will be solved as every developer will use the common mysql image.

Comment (j4rs) [reply .gitignore:20]: Exactly, you got it, it is the idea of using docker, to standardize the setup and dev environment even making it equal or close to production.

**Comment by j4rs** [comment]
@prajaktar, another question: Why is `products` repeated as a nested resource of `categeories` (only for _download_ action) and then as a standalone (for the rest of the actions)?

**Comment by prajaktar** [comment]
So here I wanted to provide a functionality where user can download list of products based on category.  As products is nested resource of categories, download action will expect a category_id which with the help of association get product details.
 
Standalone route for products is to list all products irrespective of category.

We can also create index, edit and update in nested resource. But to show example of nested and standalone routes I have added this feature.

**Comment by j4rs** [comment]
@prajaktar I think it is a better idea to continue the renewing opening issues, that way we can track it and if you send changes we can mark them as solved :), so will continue there.

**Comment by prajaktar** [comment]
Sure that we will be great. Do you want me to work on nested routes? 
You can create issue and I will work on it.

**Pull request merged**